app/Helpers/Bank/Nordigen/Transformer/AccountTransformer.php

```diff
@@ -0,0 +1,121 @@
+                },
+                {
+                    [balanceAmount[: {
+                        [amount] => 10325.64
+                        [currency] => EUR
+                    },
+                    [balanceType] => interimAvailable
+                    [creditLimitIncluded]: true,
+                    [referenceDate] => 2023-12-01
+                }
+            ]
+    [institution] => stdClass Object
+        (
+            [id] => STADT_KREISSPARKASSE_LEIPZIG_WELADE8LXXX
+            [name] => Stadt- und Kreissparkasse Leipzig
+            [bic] => WELADE8LXXX
+            [transaction_total_days] => 360
+            [countries] => [
+                "DE"
+            ],
+            [logo] => https://storage.googleapis.com/gc-prd-institution_icons-production/DE/PNG/sparkasse.png
+            [supported_payments] => {
+                [single-payment] => [
+                    "SCT",
+                    "ISCT"
+                ]
+            },
+            [supported_features] => [
+                "card_accounts",
+                "payments",
+                "pending_transactions"
+            ],
+            [identification_codes] => []
+        )
+
+    )
+ */
+
+
+class AccountTransformer implements AccountTransformerInterface
+{
+
+    public function transform($nordigen_account)
+    {
+
+        if (!property_exists($nordigen_account, 'data') || !property_exists($nordigen_account, 'metadata') || !property_exists($nordigen_account, 'balances') || !property_exists($nordigen_account, 'institution'))
+            throw new \Exception('invalid dataset');
+
+        $used_balance = $nordigen_account->balances[0];
+        // prefer entry with closingBooked
+        foreach ($nordigen_account->balances as $entry) {
+            if ($entry["balanceType"] === 'closingBooked') { // available: closingBooked, interimAvailable
+                $used_balance = $entry;
+                break;
+            }
+        }
+
+        return [
+            'id' => $nordigen_account->metadata["id"], // TODO: maybe add prefix for unique id between yodlee and nordigen?
+            'account_type' => "bank_account", // TODO: not creditCard, which type should be used here?!
+            'account_name' => $nordigen_account->data["iban"],
+            'account_status' => $nordigen_account->metadata["status"],
+            'account_number' => '**** ' . substr($nordigen_account->data["iban"], -7),
+            'provider_account_id' => $nordigen_account->data["iban"],
+            'provider_id' => $nordigen_account->institution["id"],
+            'provider_name' => $nordigen_account->institution["name"],
+            'nickname' => $nordigen_account->data["ownerName"] ? $nordigen_account->data["ownerName"] : '',
+            'current_balance' => (int) $used_balance ? $used_balance["balanceAmount"]["amount"] : 0,
+            'account_currency' => $used_balance ? $used_balance["balanceAmount"]["currency"] : '',
```

Comment:
just confirming this iso 3char currency code

Reply:
yes, you are right. the response is like from the comment above (their api-reference does not show any hint about this):
[balances] => [
                {
                    [balanceAmount]: {
                        [amount] => 9825.64
                        [currency] => EUR
                    },
                    [balanceType] => closingBooked
                    [referenceDate] => 2023-12-01
                },
                {
                    [balanceAmount[: {
                        [amount] => 10325.64
                        [currency] => EUR
                    },
                    [balanceType] => interimAvailable
                    [creditLimitIncluded]: true,
                    [referenceDate] => 2023-12-01
                }
            ]